server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/CVRDownloadByCounty.java

```diff
@@ -62,57 +69,98 @@ public String endpointName() {
    */
   @Override
   public String endpoint(final Request the_request, final Response the_response) {
-    final Set<String> county_set = the_request.queryParams();
     String result = "";
     int status = HttpStatus.OK_200;
     
-    try {
-      final OutputStream os = SparkHelper.getRaw(the_response).getOutputStream();
-      final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
+    if (validateParameters(the_request)) {
+      final Set<Integer> county_set = new HashSet<Integer>();
+      for (final String s : the_request.queryParams()) {
+        try {
+          county_set.add(Integer.valueOf(s));
+        } catch (final NumberFormatException e) {
+          // cannot happen because we validated the parameters
+        }
+      }
       final Set<CastVoteRecord> matches = getMatching(county_set);
       if (matches == null) {
-        result = "Unable to fetch records from database";
         status = HttpStatus.INTERNAL_SERVER_ERROR_500;
+        result = "Error retrieving records from database";
       } else {
-        Main.GSON.toJson(matches, bw);
-        bw.flush();
-        result =  "";
+        try {
+          final OutputStream os = SparkHelper.getRaw(the_response).getOutputStream();
+          final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(os, "UTF-8"));
+
+          Main.GSON.toJson(matches, bw);
+          bw.flush();
+        } catch (final IOException e) {
+          status = HttpStatus.INTERNAL_SERVER_ERROR_500;
+          result = "Unable to stream response.";
+        }
       }
-    } catch (final IOException e) {
-      status = HttpStatus.INTERNAL_SERVER_ERROR_500;
-      result = "Unable to stream response.";
+    } else {
+      status = HttpStatus.BAD_REQUEST_400;
+      result = "invalid county ID specified";
     }
-    
     the_response.status(status);
     return result;
   }
   
+  /**
+   * Validates the parameters of a request. For this endpoint, 
+   * the paramter names must all be integers.
+   * 
+   * @param the_request The request.
+   * @return true if the parameters are valid, false otherwise.
+   */
+  private boolean validateParameters(final Request the_request) {
+    boolean result = true;
+    
+    for (final String s : the_request.queryParams()) {
+      try {
```

Comment:
Well, since we may be validating them different ways in different places, we might need many such functions; so far the validation has just been for integers but that may not end up being the case everywhere. I'll give it some more thought at some point.